fabric-mining-level-api-v1/src/main/java/net/fabricmc/fabric/api/mininglevel/v1/FabricTool.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.mininglevel.v1;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.item.ToolMaterial;
+import net.minecraft.tag.Tag;
+
+public interface FabricTool {
+	@Nullable
+	default ToolMaterial getToolMaterial() {
+		return null;
+	}
+
+	@Nullable
+	default Tag<Block> getEffectiveBlocks() {
+		return null;
+	}
+
+	default int getMiningLevel() {
+		ToolMaterial toolMaterial = getToolMaterial();
+
+		if (toolMaterial != null) {
+			return getToolMaterial().getMiningLevel();
+		} else {
+			return 0;
+		}
+	}
+
+	default boolean isSuitableFor(int miningLevel, BlockState state) {
```

Comment:
Is this called anywhere in the mining level API? Any tool attributes stuff should stay in that module - all features here should work without the tool attribute api being present.

Reply:
While it is only called by the tool attributes module, it is necessary for that module's `DynamicAttributeTool`s to work with dynamic mining levels introduced by this module.

I don't see your point exactly - it only being called by another module in no way means that this module relies on the other. Mining Level will work just fine without Tool Attributes.

Reply:
My point is that the interface has features that don't work when the module is used alone with just its deps. I guess you could work around that by saying something like "Called in other Fabric API modules" in the JD as an explanation for it, but that feels a bit hacky to me.

Reply:
> the interface has features that don't work when the module is used alone with just its deps.
What? The feature of the interface works just fine, it still returns exactly as it should.
This method doesn't require tool attributes at all, it's the other way around. Tool attributes requires this method. This method works just fine without tool attributes.

Reply:
You'd expect this method to at least do something when overridden, when it does precisely nothing since it's not called. I don't really see why the usual item method doesn't suffice here since you also know the mining level from the same interface.

Reply:
That's... actually a good point, don't know why I didn't consider that.

I've updated it so now this method is actually used by the regular `isSuitableFor`, and thus will work when overridden.